sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2309,13 +2309,13 @@ def get_blob_tags(self, **kwargs: Any) -> Optional[Dict[str, str]]:
             see `here <https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob
             #other-client--per-operation-configuration>`__.
         :returns: Key value pairs of blob tags.
-        :rtype: Optional[Dict[str, str]]
+        :rtype: Dict[str, str]
         """
         version_id = get_version_id(self.version_id, kwargs)
         options = _get_blob_tags_options(version_id=version_id, snapshot=self.snapshot, **kwargs)
         try:
             _, tags = self._client.blob.get_tags(**options)
-            return parse_tags(tags)
+            return cast(Dict[str, str], parse_tags(tags))
```

Comment:
For some reason, it seems that when we use `list_blobs` from `ContainerClient`, this calls into `BlobPropertiesPaged`, which calls into `_build_item` and builds the blob properties using `get_blob_properties_from_generated_code` which uses `parse_tags` which returns `None` instead of an empty dict `{}` (like it would with no tags and using `get_blob_tags` directly) RIP So need to cast here for regular `get_blob_tags` calls